lib/pliny/errors.rb

```diff
@@ -1,117 +1,64 @@
+require "i18n"
+
 module Pliny
   module Errors
     class Error < StandardError
-      attr_accessor :id
 
-      def self.render(error)
-        headers = { "Content-Type" => "application/json; charset=utf-8" }
-        data = { id: error.id, message: error.message }
-        [error.status, headers, [MultiJson.encode(data)]]
-      end
+      class << self
+        attr_accessor :error_class_id, :error_class_status
 
-      def initialize(message, id)
-        @id = id
-        super(message)
+        def render(error)
+          headers = { "Content-Type" => "application/json; charset=utf-8" }
+          data = { id: error.id, message: error.user_message }.merge(error.metadata)
+          [error.status, headers, [MultiJson.encode(data)]]
+        end
       end
-    end
 
-    class HTTPStatusError < Error
-      attr_accessor :status
+      attr_accessor :id, :status, :metadata
 
-      def initialize(message = nil, id = nil, status = nil)
-        meta    = Pliny::Errors::META[self.class]
-        message = message || meta[1] + "."
-        id      = id || meta[1].downcase.tr(' ', '_').to_sym
-        @status = status || meta[0]
-        super(message, id)
+      def initialize(id=nil, metadata: {})
```

Comment:
This is a breaking API change which will cause me some pain in a few places. Mostly where we are raising errors with custom messages, for example:

```ruby
fail Pliny::Errors::BadRequest, MY_CUSTOM_MESSAGE
```

Again, I agree that using an error `id` and `i18n` to sort this out is a better solution long term.